java/daikon/chicory/MethodGen24.java

```diff
@@ -0,0 +1,347 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import org.checkerframework.checker.lock.qual.GuardSatisfied;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * The MethodGen24 class is a simplfied replacement for the BCEL MethodGen class. It collects and
+ * stores all the relevant information about a method.
+ */
+public class MethodGen24 {
+  /*
+   * Corresponds to the fields found in a MethodModel object.
+   */
+  // Optional<CodeModel> code() Returns the body of this method, if there is one.
+  // AccessFlags flags() Returns the access flags.
+  // Utf8Entry methodName() Returns the name of this method.
+  // Utf8Entry methodType() Returns the method descriptor of this method.
+  // default MethodTypeDesc methodTypeSymbol() Returns the method descriptor of this method, as a
+  // symbolic descriptor.
+  // Optional<ClassModel> parent() Returns the class model this method is a member of, if known.
+
+  /** The method's code model or null if no code. */
+  private @Nullable CodeModel code;
+
+  /** The method's access flags. */
+  protected AccessFlags accessFlags;
+
+  /** The method's name. */
+  private String methodName;
+
+  /** The method's type descriptor. */
+  private MethodTypeDesc mtd;
+
+  /** Whether of not the method is static. */
+  private boolean isStatic;
+
+  /** The method's CodeAttribute. */
+  private @Nullable CodeAttribute codeAttribute;
+
+  // fields of the code attribute
+  /** The method's maximum number of locals. */
+  private int maxLocals;
+
+  /** The method's maximum stack size. */
+  private int maxStack;
+
+  /** The name of the method's enclosing class. */
+  private String className;
+
+  /** The method's instruction list. */
+  private List<CodeElement> codeList;
+
+  /** The method's descriptor. */
+  private String descriptor;
+
+  /** The method's signature. */
+  private String signature;
+
+  // fields of the MethodTypeDescriptor
+  /** The method's argument types. */
+  private ClassDesc[] argTypes;
+
+  /** The method's return type. */
+  private ClassDesc returnType;
+
+  /** The method's argument names. */
+  private String[] argNames;
+
+  /** The method's local variable table. */
+  private LocalVariable[] localVariables;
+
+  /**
+   * Constructor for the MethodGen24 object.
+   *
+   * @param methodModel for the method
+   * @param className for the containing class
+   * @param inst_obj the daikon.chicory.Instrument instance
+   */
+  public MethodGen24(
+      final MethodModel methodModel, final String className, daikon.chicory.Instrument24 inst_obj) {
+
+    accessFlags = methodModel.flags();
+    methodName = methodModel.methodName().stringValue();
+    // System.out.println("methodName: " + methodName);
+    descriptor = methodModel.methodType().stringValue();
+    this.className = className;
+    isStatic = accessFlags.has(AccessFlag.STATIC);
+
+    Optional<CodeModel> code = methodModel.code();
+    if (code.isPresent()) {
```

Comment:
This code style differs from that below.

Reply:
I think this comment is  out of date.